dbms/src/Storages/DeltaMerge/DeltaMergeStore.cpp

```diff
@@ -1042,7 +1042,12 @@ void DeltaMergeStore::mergeDeltaAll(const Context & context)
 
     for (auto & segment : all_segments)
     {
-        segmentMergeDelta(*dm_context, segment, TaskRunThread::Foreground);
+        auto res = segmentMergeDelta(*dm_context, segment, TaskRunThread::Foreground);
+        // just for the bench test
+        while (res == nullptr)
+        {
+            res = segmentMergeDelta(*dm_context, segment, TaskRunThread::Foreground);
+        }
```

Comment:
just curious, there should be no concurrent task for the same segment in a bench test, why do we need this while loop?